### PR TITLE
Julient/avro module

### PR DESCRIPTION
**Pull request by jto**
Based on https://github.com/spotify/scio/pull/1096.
Implement nio for Avro

### CODE REVIEW

**Comment by nevillelyh** [comment]
Is it possible to undo the style change to make this more readabile? :panda_face:

**Comment by ravwojdyla** [comment]
fyi - will review once #1096 is merged, so it's not a cluster of all the possible changes.

**Comment by ravwojdyla** [comment]
@jto could you pls rebase this on `nio`?

**Comment by jto** [comment]
@ravwojdyla branch rebased :)

File: scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala [hunk 1]
```diff
@@ -23,11 +23,11 @@ import org.scalatest.{FlatSpec, Matchers}
 
 class TypedAvroIT extends FlatSpec with Matchers {
   "Typed Avro IO" should "not throw exception" in {
-    noException should be thrownBy {
+    // noException should be thrownBy {
```
Comment (nevillelyh) [inline scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: Is this failing? Should we add a `FIXME` or something to fix it later?

Comment (jto) [reply scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: Yeah do not merge just yet. 
I forgot about this one but I need to have a look at it...

Comment (jto) [reply scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: Fixed. The test was actually passing.

**Pull request merged**